application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -125,40 +159,84 @@ private void handleCreateCommand(SlashCommandInteractionEvent event) {
     }
 
     private void handleListCommand(SlashCommandInteractionEvent event) {
-        BiFunction<Long, Instant, String> getDescription = (channelId, remindAt) -> """
-                Channel: %s
-                Remind at: %s""".formatted(MessageUtils.mentionChannelById(channelId),
-                TimeFormat.DEFAULT.format(remindAt));
+        Result<PendingRemindersRecord> pendingReminders =
+                getPendingReminders(event.getGuild(), event.getUser());
 
-        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
-            .setColor(RemindRoutine.AMBIENT_COLOR);
+        event.reply(createListMessage(pendingReminders, 1)).setEphemeral(true).queue();
+    }
 
-        long guildId = event.getGuild().getIdLong();
-        long userId = event.getUser().getIdLong();
+    private Result<PendingRemindersRecord> getPendingReminders(Guild guild, User user) {
+        return database.read(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.GUILD_ID.eq(guild.getIdLong())
+                .and(PENDING_REMINDERS.AUTHOR_ID.eq(user.getIdLong())))
+            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
+            .fetch());
+    }
 
-        Result<PendingRemindersRecord> pendingReminders = getReminders(guildId, userId);
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = Math.ceilDiv(pendingReminders.size(), REMINDERS_PER_PAGE);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
 
         if (pendingReminders.isEmpty()) {
             remindersEmbed.setDescription("No pending reminders");
         } else {
-            pendingReminders.forEach(reminder -> {
-                String content = reminder.getContent();
-                long channelId = reminder.getChannelId();
-                Instant remindAt = reminder.getRemindAt();
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: %d/%d".formatted(pageToShow, totalPages));
 
-                remindersEmbed.addField(content, getDescription.apply(channelId, remindAt), false);
-            });
+                getPageEntries(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
```

Comment:
this is duplicated in both branches. try to get rid of that

Reply:
no as in "its not duplicated" or as in "i refuse to do that for no reasons"?

these two lines are identical and thats not good and can easily be get rid of:

![duplication](https://i.imgur.com/UmEo6r9.png)

Reply:
> its not duplicated

both are getting called on different things

Reply:
so ur point is that u dont know how to get rid of the duplication? thats fair, let me see myself then

Reply:
there u go:
```java
} else {
    List<PendingRemindersRecord> pendingRemindersForPage = pendingReminders.stream().toList();
    if (totalPages > 1) {
        remindersEmbed.setFooter("Page: %d/%d".formatted(pageToShow, totalPages));
        pendingRemindersPage.addActionRow(createPageTurnButtons(pageToShow, totalPages));

        pendingRemindersForPage = getPageEntries(pendingReminders, pageToShow);
    }

    pendingRemindersForPage.forEach(reminder -> addReminderAsField(reminder, remindersEmbed))
}
```

Reply:
yes i was thinking of something like this, y not we just change the main variable to list??

Reply:
also fine by me 👍

Reply:
done